py/jpy/src/main/c/jpy_jtype.c

```diff
@@ -423,6 +423,40 @@ int JType_CreateJavaShortObject(JNIEnv* jenv, JPy_JType* type, PyObject* pyArg,
     return JType_CreateJavaObject(jenv, type, pyArg, JPy_Short_JClass, JPy_Short_Init_MID, value, objectRef);
 }
 
+int JType_CreateJavaIntegerObjectEx(JNIEnv* jenv, JPy_JType* type, PyObject* pyArg, jobject* objectRef)
```

Comment:
I don't like the name of this method, it suggest the return type is as always a java `java.lang.Integer`.

JType_CreateJavaByteObject returns java.lang.Byte
JType_CreateJavaShortObject returns java.lang.Short
JType_CreateJavaIntegerObject returns java.lang.Integer

etc.

Reply:
Changed to JType_CreateJavaIntegerTypeObjectByInference